src/main/java/frc/robot/subsystems/intake/IntakeIOSim.java

```diff
@@ -1,53 +1,66 @@
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.Timer;
+import utils.motors.TalonFXSim;
 
 public class IntakeIOSim implements IntakeIO {
-    private final FlywheelSim spinMotor;
-    private final SingleJointedArmSim angleMotor;
-    private final PIDController angleFeedback;
+    private final TalonFXSim angleMotorSim;
+    private final TalonFXSim powerMotorSim;
+    private final TalonFXConfiguration angleConfiguration;
+    private final PIDController angleController = new PIDController(IntakeConstants.kP, IntakeConstants.kI, IntakeConstants.kD);
 
     public IntakeIOSim() {
-        spinMotor = new FlywheelSim(DCMotor.getNEO(1), IntakeConstants.SPIN_GEAR_RATIO, 1);
-        angleMotor = new SingleJointedArmSim(LinearSystemId.createSingleJointedArmSystem(
-                DCMotor.getFalcon500(1), 2, IntakeConstants.ANGLE_GEAR_RATIO
-        ), DCMotor.getFalcon500(1), IntakeConstants.ANGLE_GEAR_RATIO, 0.7, -Math.PI * 2, Math.PI * 2, false, 0); //could be wrong type of motor or max/min angle
-        angleFeedback = new PIDController(IntakeConstants.kP, IntakeConstants.kI, IntakeConstants.kD);
-    }
+        angleConfiguration = new TalonFXConfiguration();
 
-    @Override
-    public void updateInputs(IntakeLoggedInputs inputs) {
-        inputs.spinMotorPower = inputs.setpointSpinMotorPower;
-        inputs.spinMotorCurrent = spinMotor.getCurrentDrawAmps();
-        inputs.angleMotorAngle = angleMotor.getAngleRads();
-        inputs.angleMotorcurrent = angleMotor.getCurrentDrawAmps();
+        angleMotorSim = new TalonFXSim(1, IntakeConstants.ANGLE_GEAR_RATIO, 0.00001);
+        powerMotorSim = new TalonFXSim(1, 1, 0.00001);
+        angleMotorSim.setController(angleController);
 
-        spinMotor.update(0.02);
-        angleMotor.update(0.02);
+        angleConfiguration.Slot0 = IntakeConstants.PIDGains;
+//        angleMotorSim.configure(angleConfiguration);
     }
 
+
     @Override
     public void setSpinMotorPower(double power) {
-        spinMotor.setInputVoltage(power * 12);
+        powerMotorSim.setControl(new DutyCycleOut(power));
     }
 
     @Override
-    public void setAngleMotorAngle(double angle) {
-        double angleMotorAppliedVoltage = angleFeedback.calculate(angleMotor.getAngleRads(), angle);
-        angleMotor.setInputVoltage(angleMotorAppliedVoltage);
+    public void setAngleMotorAngle(Rotation2d angle) {
+        angleMotorSim.setControl(new PositionVoltage(angle.getRotations()));
     }
 
     @Override
     public void setAngleMotorPower(double power) {
-        angleMotor.setInputVoltage(power * 12);
+        angleMotorSim.setControl(new DutyCycleOut(power));
     }
 
+
     @Override
     public void resetEncoder(double angle) {
 
     }
+
+    @Override
+    public void updateInputs(IntakeLoggedInputs inputs) {
+        //var batterySim = new BatterySim();
+        angleMotorSim.update(Timer.getFPGATimestamp());
+        powerMotorSim.update(Timer.getFPGATimestamp());
```

Comment:
Does `update` receive the time since the last call or the system clock since the system started?
Also, I'm not sure the function you used is available in simulation.